mlx/backend/common/compiled_cpu.cpp

```diff
@@ -102,11 +193,30 @@ void* compile(
     source_file << source_code;
     source_file.close();
 
-    std::ostringstream build_command;
-    build_command << "g++ -std=c++17 -O3 -Wall -fPIC -shared '"
-                  << source_file_path << "' -o '" << shared_lib_path << "'";
-    std::string build_command_str = build_command.str();
-    auto return_code = system(build_command_str.c_str());
+#ifdef _MSC_VER
+    const VisualStudioInfo& info = GetVisualStudioInfo();
+    std::string libpaths;
+    for (const std::string& lib : info.libpaths) {
+      libpaths += fmt::format(" /libpath:\"{0}\"", lib);
+    }
+    std::string build_command = fmt::format(
+        "\""
+        "cd /D \"{0}\" && "
+        "\"{1}\" /LD /EHsc /MD /Ox /nologo /std:c++17 \"{2}\" "
+        "/link /out:\"{3}\" {4} >nul"
+        "\"",
+        std::filesystem::temp_directory_path().string(),
+        info.cl_exe,
+        source_file_name.str(),
+        shared_lib_name.str(),
+        libpaths);
+#else
+    std::string build_command = fmt::format(
+        "g++ -std=c++17 -O3 -Wall -fPIC -shared '{0}' -o '{1}'",
+        source_file_path,
+        shared_lib_path);
+#endif
+    auto return_code = system(build_command.c_str());
```

Comment:
I think it would be good to encapsulated the compiler into it's own class to contain the if-defs and make this more maintainable / extensible.

Something like:

```c++
static std::string JitCompiler::build_command(
  const std::string& source_file_path, const std::string& shared_lib_name)

// And we could include any special source code headers needed in the class as well. 
static std::string JitCompiler::get_header()
```

Reply:
I think it is good timing to move the code to a separate file, I have updated the PR.